packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx

```diff
@@ -93,21 +93,28 @@ export const _TagsSurvey = (props: _TagsSurveyProps): JSX.Element => {
 
   const [checkedTextFields, setCheckedTextFields] = useState<string[]>([]);
 
-  const tags: _SurveyTag[] = useMemo(() => {
-    const tags: _SurveyTag[] = [];
-    Object.keys(callIssuesToTag).forEach((issueCategory) => {
-      Object.keys(callIssuesToTag[issueCategory]).map((issue) => {
-        const issueCapitalized = issue?.charAt(0).toUpperCase() + issue?.slice(1);
-
+  const tags: _SurveyTag = useMemo(() => {
+    const tags: _SurveyTag = {};
+    Object.keys(callIssuesToTag).forEach((category) => {
+      const issueCategory = category as keyof SurveyIssues;
```

Comment:
while this is a tough one we are losing the value in strongly typing by just using `as`, let me know if you want to work together on this one.

We'll likely need to change type higher up, e.g. I believe _SurveyTag should be something more like:
```diff
-export type _SurveyTag = {
-  [issueCategory: string]: {
-    message: string;
-    issue: _AudioIssue | _OverallIssue | _ScreenshareIssue | _VideoIssue;
-  }[];
-};

+export type _SurveyTag = Record<
+  _IssueCategory,
+  {
+    message: string;
+    issue: _AudioIssue | _OverallIssue | _ScreenshareIssue | _VideoIssue;
+  }[]
+>;
```

Reply:
Gotcha! The pattern is makings sense now. Wasn't aware that we should be avoiding 'as' a as much as possible in general. Thanks for going through and providing all these great suggestions and tips and tricks! :D